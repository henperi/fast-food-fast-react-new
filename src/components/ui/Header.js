import React from "react";
import { NavLink, Link } from "react-router-dom";

const Header = () => (
  <header>
    <nav className="topnav" id="myTopnav">
      <div className="left-nav">
        <Link to="/" className="brand">
          Fast Food Fast
        </Link>
      </div>
      <div className="right-nav">
        <NavLink to="/users/foods" className="">
          Foods Menu
        </NavLink>
        <NavLink to="#about">About Us</NavLink>
        <NavLink to="#contact">Contact Us</NavLink>
        <NavLink to="/signin" className="">
          Signin
        </NavLink>
        <NavLink to="/signup">Signup</NavLink>
        <NavLink className="btn btn-green" to="/users/my-cart">
          My Cart
          <i className="fa fa-shopping-cart" />{" "}
          <span className="count cart-count">0</span>
        </NavLink>
        <a href="javascript:void(0);" className="icon">
          &#9776;
        </a>
      </div>
    </nav>
  </header>
);

export default Header;
