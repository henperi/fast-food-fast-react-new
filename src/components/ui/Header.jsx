import React, { PureComponent, Fragment } from 'react';
import { NavLink, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser } from '../../actions/userActions';

class Header extends PureComponent {
  handleLogout = (event) => {
    event.preventDefault();

    const { logoutUser: dispatchLogout, history, user } = this.props;
    dispatchLogout();

    localStorage.removeItem('userToken');
    localStorage.removeItem('userId');
    localStorage.removeItem('fullname');
    localStorage.removeItem('role');
  };

  toggleNav = (event) => {
    event.preventDefault();
  };

  toggleNav = (event) => {
    event.preventDefault();
    return (
      event.target.parentElement.parentElement.classList.contains('topnav')
      && event.target.parentElement.parentElement.classList.toggle('responsive')
    );
  };

  render() {
    const { user, cart } = this.props;
    const cartCount = cart.length;

    const IndexLinks = `${window.origin}/` === window.location.href && (
      <Fragment>
        <NavLink to="#about">About Us</NavLink>
        <NavLink to="#contact">Contact Us</NavLink>
      </Fragment>
    );

    const guestNav = (
      <div className="right-nav">
        <NavLink to="/users/foods" className="">
          Foods Menu
        </NavLink>
        {IndexLinks}

        <NavLink to="/signin" className="">
          Signin
        </NavLink>
        <NavLink to="/signup">Signup</NavLink>
        <NavLink className="btn btn-green" to="/users/my-cart">
          My Cart
          <i className="fa fa-shopping-cart" />
          {' '}
          <span className="count cart-count">{cartCount}</span>
        </NavLink>
        <a
          onClick={(e) => {
            this.toggleNav(e);
          }}
          href="/"
          className="icon"
        >
          &#9776;
        </a>
      </div>
    );
    const userNav = (
      <div className="right-nav">
        <NavLink to="/users/foods" activeClassName="active">
          Foods Menu
        </NavLink>
        {IndexLinks}
        <div className="dropdown">
          <NavLink to="/users/my-orders" activeClassName="active">
            My Orders
          </NavLink>

          {/*
            <div className="dropdown-content">
            <NavLink to="/pending-orders">Pending Orders</NavLink>
            <NavLink to="/completed-orders">Completed Orders</NavLink>
            <NavLink to="/all-orders">All My Orders</NavLink>
            </div>
          */}
        </div>
        {/*        <NavLink to="/my-profile" className="">
          My Profile
        </NavLink> */}
        <NavLink className="btn btn-green" to="/users/my-cart">
          My Cart
          <i className="fa fa-shopping-cart" />
          {' '}
          <span className="count cart-count">{cartCount}</span>
        </NavLink>
        <a onClick={this.handleLogout} href="/logout" className="btn-rounded logout">
          Logout
        </a>
        <a
          onClick={(e) => {
            this.toggleNav(e);
          }}
          href="/"
          className="icon"
        >
          &#9776;
        </a>
      </div>
    );
    const adminNav = (
      <div className="right-nav">
        <NavLink to="/admins/foods" className="active">
          Foods Menu
        </NavLink>
        {IndexLinks}
        <div className="dropdown">
          <button type="button" className="dropbtn">
            Orders
            <i className="fa fa-caret-down" />
          </button>
          <div className="dropdown-content">
            <NavLink to="/admins/pending-orders">Pending Orders</NavLink>
            <NavLink to="/admins/completed-orders">Completed Orders</NavLink>
            <NavLink to="/admins/all-orders">All Orders</NavLink>
          </div>
        </div>
        <NavLink to="/admins/my-profile" className="">
          My Profile
        </NavLink>
        <a onClick={this.handleLogout} href="/" className="btn-rounded logout">
          Logout
        </a>
        <a
          onClick={(e) => {
            this.toggleNav(e);
          }}
          href="/"
          className="icon"
        >
          &#9776;
        </a>
      </div>
    );

    let nav = guestNav;
    if (user.isAuthenticated) {
      nav = user.role === 'admin' ? adminNav : userNav;
    }
    return (
      <header>
        <nav className="topnav" id="myTopnav">
          <div className="left-nav">
            <Link to="/" className="brand">
              Fast Food Fast
            </Link>
          </div>
          {nav}
        </nav>
      </header>
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
  cart: state.cart,
});

export default connect(
  mapStateToProps,
  { logoutUser },
)(Header);
