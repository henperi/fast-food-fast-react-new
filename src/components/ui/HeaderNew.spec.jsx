/* eslint-disable no-undef */
/* eslint-disable react/react-in-jsx-scope */
import { shallow } from 'enzyme';
import { Fragment } from 'react';
import { Header } from './Header';

describe('rendering', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      logoutUser: jest.fn(),
      errors: [],
      cart: [],
      history: {
        push: jest.fn(),
      },
      user: {
        isAuthenticated: true,
      },
    };
  });

  it('should render the component', () => {
    wrapper = shallow(<Header {...props} />);
    expect(wrapper.find('Fragment')).toBeDefined();
    expect(wrapper.find('Fragment')).toHaveLength(0);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
  });

  // it('should test mapStateToProps', () => {
  //   const state = mapStateToProps(mockedState);
  //   expect(state).toBeTruthy();
  // });

  it('should test the handleLogout function', () => {
    const fakeEvent = { preventDefault: () => console.log('preventDefault') };
    wrapper = shallow(<Header {...props} />);
    const logoutBtn = wrapper.find('.logout');
    logoutBtn.simulate('click', fakeEvent);
    const logoutUser = jest.fn();
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().handleLogout);
    });
    promise.then(() => expect(logoutUser).toHaveBeenCalledTimes(1));
  });

  it('should test the toggleNav function', () => {
    // event.target.parentElement.parentElement.classList.toggle;
    const fakeEvent = {
      preventDefault: () => console.log('preventDefault'),
      target: {
        parentElement: { parentElement: { classList: { toggle: jest.fn(), contains: jest.fn() } } },
      },
    };
    wrapper = shallow(<Header {...props} />);
    const toggleBtn = wrapper.find('.icon');
    toggleBtn.simulate('click', fakeEvent);
    const toggleNav = jest.fn(fakeEvent);
    const promise = new Promise((resolve) => {
      resolve(wrapper.instance().handleLogout);
    });
    promise.then(() => expect(toggleNav).toHaveBeenCalledTimes(1));
  });
});
