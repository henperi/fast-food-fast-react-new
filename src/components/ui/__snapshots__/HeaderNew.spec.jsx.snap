// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render the component 1`] = `
<header>
  <nav
    className="topnav"
    id="myTopnav"
  >
    <div
      className="left-nav"
    >
      <Link
        className="brand"
        replace={false}
        to="/"
      >
        Fast Food Fast
      </Link>
    </div>
    <div
      className="right-nav"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/users/foods"
      >
        Foods Menu
      </NavLink>
      <div
        className="dropdown"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/users/my-orders"
        >
          My Orders
        </NavLink>
      </div>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="btn btn-green"
        to="/users/my-cart"
      >
        My Cart
        <i
          className="fa fa-shopping-cart"
        />
         
        <span
          className="count cart-count"
        >
          0
        </span>
      </NavLink>
      <a
        className="btn-rounded logout"
        href="/logout"
        onClick={[Function]}
      >
        Logout
      </a>
      <a
        className="icon"
        href="/"
        onClick={[Function]}
      >
        â˜°
      </a>
    </div>
  </nav>
</header>
`;
